### ISSUE TITLE
Data not fetched when using partial render policy in 10.1.0

### ISSUE THREAD

**Issue by janicduplessis** [facebook/relay#3270]
Since updating to 0.0.0-experimental-c818bac3 from 0.0.0-experimental-183bdd28 I'm getting missing data and a `
Relay: Tried reading fragment [...], but it has missing data [...]` warning.

I have something similar to this:

```js
const query = useLazyLoadQuery(
    graphql`
      query Query($id: ID!) {
        node(id: $id) {
          ... FragmentA_a
          ... FragmentB_b
        }
      }
    `,
    { id },
    { fetchPolicy: 'store-and-network', UNSTABLE_renderPolicy: 'partial' },
  );

const a = useFragment(
    graphql`
      fragment FragmentA_a on A {
        id
      }
    `,
    query.a,
  );

<Suspense>
  <FragmentB b={query.b} />
</Suspense>
```

If the data for `FragmentA_a` is available but missing for `FragmentB_b`  then the `FragmentB` component still tries to render with missing data instead of suspending. This used to work with the previous experimental release. Note that the issue seems to be in the react-relay package since using 10.1.0 runtime with older 0.0.0-experimental-183bdd28 release seem to work fine.

**Comment by maraisr** [comment]
Think this is related to: https://github.com/facebook/relay/issues/3215

**Comment by alunyov** [comment]
cc @jstejada

**Comment by jstejada** [comment]
hey @janicduplessis, sorry for the delay here, this definitely sounds off. quick question: which versions of react and relay were you using? were you using react concurrent mode?

**Comment by janicduplessis** [comment]
@jstejada Hey! No worries, this is using react@0.0.0-experimental-3310209d0 with react-native@baa824826dce30cc69767503291860ad63eb063c so that would be no concurrent mode I think.

**Comment by jstejada** [comment]
hey @janicduplessis, this should be fixed with 24bac58f46bc7694f069e46c4493201dfea60786, if you keep seeing issues feel free to reopen

**Comment by janicduplessis** [comment]
Thanks! Actually I just tested updating to `0.0.0-experimental-4c4107dd` and the issue does not happen so looks like something else was the issue.
